std/heap.zig

```diff
@@ -170,16 +166,20 @@ pub const DirectAllocator = struct {
                 ) orelse return error.OutOfMemory;
                 const offset = old_adjusted_addr - root_addr;
                 const new_root_addr = @ptrToInt(new_ptr);
-                const adjusted_addr = new_root_addr + offset;
-                const new_adjusted_addr = mem.alignForward(new_root_addr, new_align);
-                // If HeapReAlloc didn't happen to move the memory to the new alignment
-                // then we need to copy it
-                if (new_adjusted_addr != adjusted_addr) {
+                var new_adjusted_addr = new_root_addr + offset;
+                const offset_is_valid = new_adjusted_addr + new_size + @sizeOf(usize) <= new_root_addr + amt;
+                const offset_is_aligned = new_adjusted_addr % new_align == 0;
+                if (!offset_is_valid or !offset_is_aligned) {
```

Comment:
I'm not confident in the style of this conditional. Suggestions for how to improve it are welcome.

Reply:
It looks fine to me - what are you concerned about?

Reply:
If it looks fine to you then there's no issue. Wasn't sure about pulling out the conditions into variables.

Reply:
Yep, in general I think that's a great technique for clarity.